### DISCUSSION TITLE
Keep shared boundaries after polygon simplification

### DISCUSSION THREAD

**Discussion by wipfli** [onthegomap/planetiler#644] [Q&A]
Is it possible to simplify polygon boundaries such that they stay shared between neighboring polygons?

I was playing around with landcover polygons and found that douglas peuker makes boundaries different for neighboring polygons. This creates gaps.

**Comment by msbarry** [comment]
The first problem is detecting that they are shared between polygons. To do it right we'd the input to be in a format like [topojson](https://github.com/topojson/topojson) or at least convey which points of a polygons are shared between more than two polygons so we can avoid removing them during simplification. Absent that though we could have a routine do the work to compute shared edges before simplification - except we'd need access to both geometries when simplifying but planetiler only processes individual geometries in isolation so we'd need to do it in post-process. 

Another option would be to limit simplification to only remove points that increase the area of a polygon but not decrease it (so it only produces overlaps but not gaps). Would that work?

**Reply by bdon** [reply]
The overlapping workaround might not be acceptable if the polygons are displayed with alpha < 1

**Reply by wipfli** [reply]
The size increase would work for my use-case. Thanks for sharing this idea!

**Reply by wipfli** [reply]
Here is an example of the defect in see: https://wipfli.github.io/median-filter-landcover/#map=9.79/9.6344/104.8832

**Reply by msbarry** [reply]
I'd also be curious what the output and file size looks like if you set simplify tolerance to  0 during the process step. That would only simplify geometries by rounding to tile coordinates. That needs to look ok for the "topology-aware simplification in post-process" approach to be a viable option.

**Comment by bdon** [comment]
Tippecanoe has this functionality with `--detect-shared-borders` 

https://github.com/felt/tippecanoe#attempts-to-improve-shared-polygon-boundaries
